src/shared/ui/components/InstructionsTable.vue

```diff
@@ -0,0 +1,261 @@
+import type { Instruction } from '@/shared/api/schemas';
+import type { useTable } from '@/shared/lib/table';
+import { computed } from 'vue';
+
+const props = defineProps<{
+  table: ReturnType<typeof useTable<Instruction>>
+  accounts?: boolean
+  allTypes?: boolean
+  hashType: 'short' | 'full'
+}>();
+
+const table = computed(() => props.table);
+</script>
+
+<template>
+  <div
+    class="instructions-table"
+    :class="{ 'instructions-table_short': props.accounts }"
+  >
+    <BaseTable
+      :loading="table.loading.value"
+      :pagination="table.pagination"
+      :items="table.items.value"
+      container-class="instructions-table__container"
+      :pagination-breakpoint="1441"
+      @next-page="table.nextPage()"
+      @prev-page="table.prevPage()"
+      @set-page="table.setPage($event)"
+      @set-size="table.setSize($event)"
+    >
+      <template #row="{ item }">
+        <div class="instructions-table__row">
+          <TransactionStatus
+            type="tooltip"
+            class="instructions-table__icon"
+            :committed="item.transaction_status === 'Committed'"
+          />
+
+          <div class="instructions-table__column">
+            <div class="instructions-table__label">
+              {{ $t('transactions.transactionID') }}
+            </div>
+
+            <BaseHash
+              :hash="item.transaction_hash"
+              :type="hashType"
+              :link="`/transactions/${item.transaction_hash}`"
+              copy
+            />
+
+            <div class="instructions-table__time">
+              {{ defaultFormat(item.created_at) }}
+            </div>
+          </div>
+
+          <div class="instructions-table__columns">
+            <div
+              v-if="props.allTypes"
+              class="instructions-table__column"
+            >
+              <div class="instructions-table__label">
+                {{ $t('kind') }}
+              </div>
+
+              <span class="row-text">{{ item.kind }}</span>
+            </div>
+
+            <div class="instructions-table__column">
+              <div class="instructions-table__label">
+                {{ $t('entity') }}
+              </div>
+
+              <span class="row-text">{{ Object.entries(item.payload)[0][0] }}</span>
+            </div>
+
+            <div class="instructions-table__column">
+              <div class="instructions-table__label">
+                {{ $t('transactions.block') }}
+              </div>
+
+              <BaseLink :to="`/blocks/${item.block}`">
+                {{ item.block }}
+              </BaseLink>
+            </div>
+          </div>
+
+          <div
+            v-if="!props.accounts"
+            class="instructions-table__column-value"
+          >
+            <div class="instructions-table__label">
+              {{ $t('value') }}
+            </div>
+
+            <span class="row-text">{{ Object.entries(item.payload)[0][1] }}</span>
```

Comment:
What does this magic mean? What does `accounts` prop mean? Impossible to guess if not looking at the invocation.

(and I don't see it even after looking at `<TransactionsList>`)

What could be helpful:

- clearer naming of props, for "dummies" (like me!)
- make a computed for this magic payload manipulation with some name, like `assuminglyAccountIdFromPayload` or idk